gentoo specific hacks because we unbundle the
deps of irrlicht

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 09edd58..1f51a30 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,7 +45,6 @@ add_subdirectory("${PROJECT_SOURCE_DIR}/lib/enet")
 include_directories("${PROJECT_SOURCE_DIR}/lib/enet/include")
 
 # Build the irrlicht library
-add_subdirectory("${PROJECT_SOURCE_DIR}/lib/irrlicht")
 include_directories("${PROJECT_SOURCE_DIR}/lib/irrlicht/include")
 
 # Build the Wiiuse library
@@ -137,6 +136,13 @@ else()
 endif()
 
 
+# Irrlicht
+if (APPLE)
+    set(IRRLICHT_LIBRARY "/Library/Frameworks/IrrFramework.framework")
+else()
+    set(IRRLICHT_LIBRARY ${PROJECT_SOURCE_DIR}/lib/irrlicht/lib/Linux/libIrrlicht.a)
+endif()
+
     
 # Set some compiler options
 if(UNIX)
@@ -226,12 +232,17 @@ target_link_libraries(supertuxkart
     bulletcollision
     bulletmath
     enet
-    stkirrlicht
     ${CURL_LIBRARIES}
+    ${IRRLICHT_LIBRARIES}
     ${OGGVORBIS_LIBRARIES}
+    ${IRRLICHT_LIBRARY}
     ${IRRLICHT_XF86VM_LIBRARY}
     ${OPENAL_LIBRARY}
-    ${OPENGL_LIBRARIES})
+    ${OPENGL_LIBRARIES}
+    png
+    jpeg
+    bz2
+    z)
 
 if(APPLE)
     # In theory it would be cleaner to let CMake detect the right dependencies. In practice, this means that if a OSX user has
